src/test/java/de/tum/in/www1/artemis/DatabaseQueryCountTest.java

```diff
@@ -76,4 +86,78 @@ void testGetAllCoursesForDashboardRealisticQueryCount() throws Exception {
         // 2 DB calls to get the quiz batches for active quiz exercises
         // 1 DB call to get all plagiarism cases
     }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student2", roles = "USER")
+    void testStartWorkingOnExamQueryCount() throws Exception {
+        Exam exam = createActiveExam();
+        StudentExam studentExam = createStudentExamForUser(exam, TEST_PREFIX + "student2");
+
+        assertThatDb(() -> {
+
+            startWorkingOnExam(studentExam);
+            return null;
+
+        }).hasBeenCalledAtMostTimes(getStartWorkingOnExamExpectedTotalQueryCount());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student3", roles = "USER")
+    void testSubmitAnswerOfExamQueryCount() throws Exception {
+        Exam exam = createActiveExam();
+        StudentExam studentExam = createStudentExamForUser(exam, TEST_PREFIX + "student3");
+        startWorkingOnExam(studentExam);
+
+        assertThatDb(() -> {
+
+            submitExam(studentExam);
+            return null;
+
+        }).hasBeenCalledAtMostTimes(getSubmitAnswerOfExamExpectedTotalQueryCount());
+    }
+
+    private Exam createActiveExam() {
+        Course course = courseUtilService.addEmptyCourse();
+        ZonedDateTime visibleDate = now().minusMinutes(10);
+        ZonedDateTime startDate = now().minusMinutes(5);
+        ZonedDateTime endDate = now().plusMinutes(5);
+        return examUtilService.addExam(course, visibleDate, startDate, endDate);
+    }
```

Comment:
Consider moving this into `ExamUtilService`

Reply:
The method is moved into ExamUtilService.